### PR TITLE
Tests for setup plugin errors

### PR DESCRIPTION
**Pull request by calvn**
Also refactored `NewTestCluster` to be able to create a cluster with an arbitrary number of cores.

### CODE REVIEW

File: vault/logical_system_integ_test.go [hunk 1]
```diff
@@ -220,6 +273,28 @@ func testSystemBackendMock(t *testing.T, numMounts int, backendType logical.Back
 	return cluster
 }
 
+func ensureCoresSealed(t *testing.T, c *vault.TestCluster) {
```
Comment (jefferai) [inline vault/logical_system_integ_test.go]: Can we maybe not have this function copied around somewhere yet again?

Let's start migrating such utility functions to a helper/testutil or so package.

File: vault/testing.go [hunk 2]
```diff
@@ -335,11 +335,17 @@ func TestAddTestPlugin(t testing.T, c *Core, name, testFunc string) {
...
+	// Determine plugin directory path
+	fullPath, err := filepath.EvalSymlinks(os.Args[0])
 	if err != nil {
 		t.Fatal(err)
 	}
-	c.pluginCatalog.directory = filepath.Dir(c.pluginCatalog.directory)
+	directoryPath := filepath.Dir(fullPath)
+
+	// Set core's plugin directory and plugin catalog directory
+	c.pluginDirectory = directoryPath
```
Comment (jefferai) [inline vault/testing.go]: Why do we have both of these?

Comment (calvn) [reply vault/testing.go]: `setupPluginCatalog()` uses `c.pluginDirectory`. If there is a unseal process during the test, the plugin loads from an empty directory and plugin mount fails:

"core: failed to create mount entry: path=mock-0/ error=no plugin found with name: mock-plugin"

**Comment by calvn** [comment]
@briankassouf I think it's because this is not a merge to master. Travis should run when I open the PR on `f-setup-plugins` and merge against master :)

**Comment by briankassouf** [comment]
@calvn Ah, didn't notice that 👍

File: vault/testing.go (hunk 2 above)
Comment (jefferai) [reply vault/testing.go]: Gotcha. 👍

File: vault/logical_system_integ_test.go (hunk 1 above)
Comment (calvn) [reply vault/logical_system_integ_test.go]: Moved to testing.go since its a `TestCluster` operation

**Comment by calvn** [comment]
@jefferai I also changed `c.Cleanup()` to use `ensureCoresSealed()` since it was pretty much the same code. I settled on a timeout of 10 seconds, but I can bump it back up to 60 if that's too short.

**Comment by jefferai** [comment]
I'd bump it back up. Other testing code (especially on ent side) can take quite a bit longer to finish up tasks before sealing.

**Comment by calvn** [comment]
Bumped, merging if no objections.

**Pull request merged**